utils/vault.py

```diff
@@ -91,12 +99,24 @@ def _read_all_v2(self, path, version):
         if secret is None or 'data' not in secret \
                 or 'data' not in secret['data']:
             raise SecretNotFound(path)
-        return secret['data']['data']
+
+        data = secret['data']['data']
+        self._cache[cache_key] = data
+        return data
 
     def _read_all_v1(self, path):
+        cache_key = path
+        cache_value = self._cache.get(cache_key)
+        if cache_value:
+            logging.debug('Vault v1 cache hit')
+            return cache_value
```

Comment:
can we really cache in v1? content will change, key wont...

Reply:
we only cache for the current execution

Reply:
an instance of this class will be used in many executions... we are a singletown now

Reply:
ah crap, that's true.

Reply:
wait, cache for v2 is still important... if the content changes, the version also changes there

Reply:
if the data is kept across executions i don't trust it so much...

Reply:
can the data change in a v2 secret while not changing the version?

Reply:
not really, but a version can be deleted

Reply:
that's fine, I guess

Reply:
Not awesome but in our use case, removing a secret version and not updating app-interface would be silly